### ISSUE TITLE
Unable to upload an image by a celery task in fastapi

### ISSUE THREAD

**Issue by alikemalagirman** [tiangolo/fastapi#5911]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
##app.py
@app.post("/upload3")
async def upload(file: UploadFile = File(...),):
    
    result=tasks.create_image.delay(file)
    url=result.get()
    return {"message":"ok"}

##tasks.py
@shared_task
async def create_image(file:_fastapi.UploadFile):
    
    urlFile = cloudinary.uploader.upload(file.file)
    url = urlFile.get("url")    

    return url
```

### Description

It throws an error as Object of type UploadFile is not JSON serializable

### Operating System

Linux

### FastAPI Version

0.85.0 

### Python Version

3.8.10

**Comment by FaridLU** [comment]
Hi @alikemalagirman, 

FYI, I am not an expert on Fast API, but what I can say according to the error (Object of type UploadFile is not JSON serializable) that you are getting, is getting triggered in the following line:

`result=tasks.create_image.delay(file)`

The reason for this is, you are trying to send a file into a celery task. Instead of passing the file from the above line, you can save the File into DB model, and then send the instance ID (instead of the file) to the celery function (which is named as create_image). 

I hope this solves your issue.

**Comment by ThirVondukr** [comment]
Yep, but more likely in production environment you'd have to store it somewhere else - just on disk, if you-re running both fastapi and celery on the same server, and in S3 or similar storage otherwise

**Comment by alikemalagirman** [comment]
Hi, @FaridLU and @ThirVondukr,
Thank you for your comments,

I want celery to complete uploading an image to a cloud storage at the background. I understand the root cause of problem. How can I do it efficiently?

**Comment by ThirVondukr** [comment]
Why use celery here? 🤔 You can do that in your endpoint directly. If it really takes a long time maybe look into `BackgroundTasks`

**Comment by alikemalagirman** [comment]
Hi, @ThirVondukr :

I want to result back.

I solved the issue,

I was due to asycn function definition.

task sholud be defined as below:

`@shared_task
def create_image(file:_fastapi.UploadFile):
    
    urlFile = cloudinary.uploader.upload(file.file)
    url = urlFile.get("url")    

    return url`
